src/job_cache/job_cache.cpp

```diff
@@ -167,28 +167,29 @@ wcl::optional<wcl::unique_fd> try_connect(std::string dir) {
   return wcl::make_some<wcl::unique_fd>(std::move(socket_fd));
 }
 
-Cache::Cache(std::string dir, uint64_t max, uint64_t low) {
-  mkdir_no_fail(dir.c_str());
-
-  // launch the daemon
-  if (daemonize(dir.c_str())) {
-    // We are the daemon, launch the cache
-
+// Launch the job cache daemon
+wcl::optional<ConnectError> Cache::launch_daemon() {
+  // We are the daemon, launch the cache
+  if (daemonize(cache_dir.c_str())) {
     std::string job_cache = wcl::make_canonical(find_execpath() + "/../bin/job-cache");
-    std::string low_str = std::to_string(low);
-    std::string max_str = std::to_string(max);
-    execl(job_cache.c_str(), "job-cached", dir.c_str(), low_str.c_str(), max_str.c_str(), nullptr);
+    std::string low_str = std::to_string(low_threshold);
+    std::string max_str = std::to_string(max_size);
+    execl(job_cache.c_str(), "job-cached", cache_dir.c_str(), low_str.c_str(), max_str.c_str(),
+          nullptr);
 
-    wcl::log::fatal("exec(%s): %s", job_cache.c_str(), strerror(errno));
+    wcl::log::error("exec(%s): %s", job_cache.c_str(), strerror(errno));
+    return wcl::make_some<ConnectError>(ConnectError::FailedToLaunchDaemon);
```

Comment:
I just realized that this error is confusing to handle since it only gets returned in the child process. Maybe this function shouldn't attempt to return an error? Like we just try to demonize and hope for the best and the other retries and logging messages will smooth user experience and debugging over?

Reply:
If it does happen though, we'll have the log, but then the child process will keep on running which is very odd.

Reply:
ohhh yeah that is pretty weird